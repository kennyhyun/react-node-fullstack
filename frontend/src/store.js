import { createStore, combineReducers, applyMiddleware, compose } from 'redux'
import thunk from 'redux-thunk';
import appReducer from './reducers/app';
import productReducer from './reducers/product';

const reducer = combineReducers({
  appReducer,
  productReducer,
});

const composeEnhancers = process.env.NODE_ENV === 'development'
  ? (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose)
  : compose;

const store = createStore(
  reducer,
  composeEnhancers(
    applyMiddleware(...[thunk])
  )
);

export default store;

