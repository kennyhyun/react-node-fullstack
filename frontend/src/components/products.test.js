import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from "react-router-dom";

import Products from './products';
import store from '../store';

it('renders correctly', () => {
  const tree = renderer
    .create(<Provider store={store}>
      <Router>
      <Products />
      </Router>
      </Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
