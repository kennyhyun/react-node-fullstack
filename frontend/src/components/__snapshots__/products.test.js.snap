// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Products-wrapper-0-1-1"
>
  <div
    className="ListHeader-wrapper-1"
  >
    <div
      className="ListHeader-rightWrapper-2"
    >
      <div
        className="MuiInputBase-root-23 MuiInput-root-10"
        onClick={[Function]}
      >
        <div
          className="MuiSelect-root-3"
        >
          <div
            aria-haspopup="true"
            aria-pressed="false"
            className="MuiSelect-select-4 MuiSelect-selectMenu-7 MuiSelect-outlined-6 MuiInputBase-input-33 MuiInput-input-18"
            id="select-8 per page"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            8 per page
          </div>
          <input
            id="items-per-page"
            name="8 per page"
            type="hidden"
            value={8}
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-40 MuiSelect-icon-9"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <h2
      className="MuiTypography-root-53 MuiTypography-h5-69"
    >
      All Products
    </h2>
    <p
      className="MuiTypography-root-53 MuiTypography-body1-62 MuiTypography-colorTextSecondary-86"
    >
      No item
    </p>
    <hr
      className="MuiDivider-root-89"
    />
  </div>
  <div
    className="MuiGrid-container-94 MuiGrid-spacing-xs-16-117"
  />
  <div
    className="ListFooter-wrapper-191"
  >
    <div
      className="ListFooter-content-192"
    >
      <button
        className="MuiButtonBase-root-223 MuiButtonBase-disabled-224 MuiButton-root-197 MuiButton-text-199 MuiButton-flat-202 MuiButton-disabled-217"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiButton-label-198"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-40"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
            />
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
          </svg>
          Previous page
        </span>
      </button>
      <ul
        className="ListFooter-pgRoot-193"
      />
      <button
        className="MuiButtonBase-root-223 MuiButtonBase-disabled-224 MuiButton-root-197 MuiButton-text-199 MuiButton-flat-202 MuiButton-disabled-217"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiButton-label-198"
        >
          Next page
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-40"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
            />
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;
